internal/controller/changetransferpolicy_controller.go

```diff
@@ -250,6 +276,7 @@ func (r *ChangeTransferPolicyReconciler) setCommitStatusState(ctx context.Contex
 				Key:   status.Key,
 				Phase: string(promoterv1alpha1.CommitPhasePending),
 			})
+			logger.Info("No CommitStatus found", "key", status.Key, "sha", targetCommitBranchState.Hydrated.Sha)
```

Comment:
Maybe just one log statement outside all the ifs? Log key, sha, phase, tooManyMatchingShas, and len(csList.Items).